tableau_utilities/scripts/cli.py

```diff
@@ -0,0 +1,120 @@
+from argparse import RawTextHelpFormatter
```

Comment:
@JustinGrilli  This `cli` get things to work the way you had requested with 1 master cli command and then subcommands

The `+` to this is that we can reuse some stuff. The `-` is that all the args have to be in a top level file.  It makes the `cli` more like an intertwined program than a collection of standalone scripts  _I'm cool with this if you are_

Here's an example usage for running the generate config script. Notice that the universal arguments are entered first before the script command

```
tableau_utilities --token_name my_token_name --token_secret 123412341234 --site mysitename  --server 10ay generate_config --datasource 'Opportunity'
```

Here's the top level -h command output
```
% tableau_utilities -h                                    
usage: tableau_utilities [-h] [--server SERVER] [--site SITE] [--api_version API_VERSION] [--user USER] [--password PASSWORD] [--token_secret TOKEN_SECRET] [--token_name TOKEN_NAME]
                         {server_info,generate_config,merge_config} ...

positional arguments:
  {server_info,generate_config,merge_config}
                        You must choose a script area to run
    server_info         Retrieve and view information from Tableau Cloud/Server
    generate_config     Generate configs to programatically manage metdatadata in Tableau datasources via Airflow
    merge_config        Merge a new config into the existing master config

options:
  -h, --help            show this help message and exit

server_information:
  Server Information

  --server SERVER       Tableau Server URL. i.e. <server_address> in https://<server_address>.online.tableau.com
  --site SITE           Site name. i.e. <site> in https://<server_address>.online.tableau.com/#/site/<site>
  --api_version API_VERSION
                        Tableau API version

user_pass:
  Authentication with username and password method

  --user USER           user name
  --password PASSWORD   password

token_info:
  Authentication with a Personal Access Token (PAT)

  --token_secret TOKEN_SECRET
                        Personal Access Token Secret
  --token_name TOKEN_NAME
                        Personal Access Token Name
```

And here's a subcommand `-h` output:
```
 % tableau_utilities generate_config -h
usage: tableau_utilities generate_config [-h] [--datasource DATASOURCE] [--clean_up_first] [--folder_name FOLDER_NAME] [--file_prefix] [--definitions_csv DEFINITIONS_CSV]

options:
  -h, --help            show this help message and exit
  --datasource DATASOURCE
                        The name of the datasources to generate a config for
  --clean_up_first      Deletes the directory and files before running
  --folder_name FOLDER_NAME
                        Specifies the folder to write the datasource and configs to
  --file_prefix         Adds a prefix of the datasource name to the output file names
  --definitions_csv DEFINITIONS_CSV
                        Allows a csv with definitions to be inputted for adding definitions to a config. It may be easier to populate definitions in a spreadsheet than in the configo

```

Reply:
@JustinGrilli  Merry Christmas!  Let me know if you want me know what you think of the ways forward here.

Reply:
@JustinGrilli  I did a little more work on this integrating some of the things from your `main` into this.   Now that I'm using this more I think it'll be a good path forward. The main CLI script you have has a lot of arguments and this isn't any different. So we'll just have to integrate all of that functionality into this.  I'll keep working on it as I use the functions.

Reply:
@JustinGrilli  This now replicates all of the functionality in your `main` script except replacing the datasource credentials and alerting columns.  

I think we can build that functionality into this but I'm running out of steam for now.